### PR TITLE
PIC-3036: enable notification banner links and html

### PR DESCRIPTION
**Pull request by paddybasi**
fixed security issue with nonce not being generated per request and implemented notification text preview

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -60,45 +60,49 @@ module.exports = function createApp ({ signInService, userService }) {
...
-        upgradeInsecureRequests: [],
-        connectSrc: [
-          '\'self\'',
-          'www.google-analytics.com',
-          'https://www.google-analytics.com',
-          'js.monitor.azure.com',
-          'dc.services.visualstudio.com'
-        ]
+  app.use((req, res, next) => {
+    res.locals.nonce = config.nonce()
```
Comment (paddybasi) [inline app.js:64]: nonce needs to be generated per request, so have reconfigured in this middleware

File: app.js [hunk 2]
```diff
@@ -60,45 +60,49 @@ module.exports = function createApp ({ signInService, userService }) {
...
+    helmet({
+      contentSecurityPolicy: {
+        directives: {
+          defaultSrc: [
+            '\'self\'',
+            'https://www.google-analytics.com',
+            'www.google-analytics.com'
+          ],
+          objectSrc: ['\'none\''],
+          frameSrc: ['https://www.youtube.com', '\'self\''],
```
Comment (paddybasi) [inline app.js:74]: added `framesrc 'self'` as it was required for notification preview

File: package.json [hunk 3]
```diff
@@ -15,8 +15,8 @@
     "npm": ">=8.0.0"
   },
   "scripts": {
-    "start-feature": "PORT=3007 NODE_ENV=development node $NODE_DEBUG_OPTION ./bin/www | bunyan -o short",
-    "start-feature:dev": "PORT=3007 NOMIS_AUTH_URL=http://localhost:9091/auth NODE_ENV=development nodemon $NODE_DEBUG_OPTION ./bin/www | bunyan -o short",
+    "start-feature": "PORT=3007 NODE_ENV=development NOTIFICATION_USERNAME=test NOTIFICATION_PASSWORD=test node $NODE_DEBUG_OPTION ./bin/www | bunyan -o short",
+    "start-feature:dev": "PORT=3007 NOMIS_AUTH_URL=http://localhost:9091/auth NODE_ENV=development NOTIFICATION_USERNAME=test NOTIFICATION_PASSWORD=test nodemon $NODE_DEBUG_OPTION ./bin/www | bunyan -o short",
```
Comment (paddybasi) [inline package.json:19]: had to set notification area to known username/password (only for integration tests) to allow tests to pass

File: app.js (hunk 2 above)
Comment (johnyfarrar) [reply app.js:74]: I think this needs cleaning up once we tidy the feature flags... Will speak to Ed about a ticket

File: server/views/set-notification.njk [hunk 4]
```diff
@@ -27,10 +26,17 @@
...
+        label: { text: "Notification banner html" },
+        attributes: {
+          "data-cy": "notification-input"
+        },
         id: "notification",
         name: "notification",
-        value: currentNotification
+        value: currentNotification,
+          errorMessage: {
+            html: "FIX THE ERROR MESSAGE BELOW.<br /><span class='govuk-!-font-size-16'>Once this is done you will be able to click the submit button.</span><br /><br /><span id='notification-error-text'></span>",
```
Comment (johnyfarrar) [inline server/views/set-notification.njk:37]: Did Patrick create this content? Just the all caps seems weird!

**Review by johnyfarrar — COMMENTED** [review]
LGTM just a couple of comments...

**Review by rschallapalli — APPROVED** [review]

**Pull request merged**